modules/apps/transfer/ibc_module.go

```diff
@@ -205,7 +203,12 @@ func (im IBCModule) OnRecvPacket(
 ) ibcexported.Acknowledgement {
 	ack := channeltypes.NewResultAcknowledgement([]byte{byte(1)})
 
-	data, ackErr := im.unmarshalPacketDataBytesToICS20V2(packet.GetData())
+	ics20Version, ok := im.keeper.GetICS4Wrapper().GetAppVersion(ctx, packet.SourcePort, packet.SourceChannel)
+	if !ok {
+		return channeltypes.NewErrorAcknowledgement(fmt.Errorf("could not retrieve app version for channel (%s, %s)", packet.SourcePort, packet.SourceChannel))
```

Comment:
I think we should use a registered error here so the error numbers are filled in (I think this string is dropped from the ack)